src/client/apps/edit/components/content/article_layouts/news.jsx

```diff
@@ -27,6 +34,26 @@ export class EditNews extends Component {
     )
   }
 
+  editSource = () => {
+    const { article, onChangeArticleAction } = this.props
+    const { isEditSourceOpen } = this.state
+    const { title } = article.news_source
+
+    return (
+      <AddSource onClick={() => { this.setState({ isEditSourceOpen: !isEditSourceOpen })}}>
+        {title && title.length ? title : 'Add Source'}
+      </AddSource>
```

Comment:
One q that maybe requires @owendodd to weigh in -- are we requiring that both a title and URL are included?  Is there such thing as a title with no link?  If we need both, or just one, we should validate this somehow.

Reply:
Title should be required, link is optional, but agree we should validate the url if one is input